libs/remote_storage/tests/test_real_s3.rs

```diff
@@ -176,22 +178,22 @@ async fn s3_delete_objects_works(ctx: &mut MaybeEnabledS3) -> anyhow::Result<()>
     let path3 = RemotePath::new(Utf8Path::new(format!("{}/path3", ctx.base_prefix).as_str()))
         .with_context(|| "RemotePath conversion")?;
 
-    let data1 = "remote blob data1".as_bytes();
+    let data1 = Bytes::from_static(b"remote blob data1");
     let data1_len = data1.len();
-    let data2 = "remote blob data2".as_bytes();
+    let data2 = Bytes::from_static(b"remote blob data2");
     let data2_len = data2.len();
-    let data3 = "remote blob data3".as_bytes();
+    let data3 = Bytes::from_static(b"remote blob data3");
     let data3_len = data3.len();
     ctx.client
-        .upload(std::io::Cursor::new(data1), data1_len, &path1, None)
+        .upload(UploadSource::Bytes(data1), data1_len, &path1, None)
         .await?;
 
     ctx.client
-        .upload(std::io::Cursor::new(data2), data2_len, &path2, None)
+        .upload(UploadSource::Bytes(data2), data2_len, &path2, None)
```

Comment:
I wonder if you could impl `From<Bytes>` as well as `From<&'static [u8]>` and `From<tokio::fs::File>` for `UploadSource`, then the diff in these tests would be way smaller (just including an .into call).

Reply:
I was hesitant to do this because it seemed a bit magic-y (if I were reading the code, I would have no context for what the `.into()` converts it to).

But that's not a strong objection, happy to make this change either way - just wanted to voice this opinion.